include/gsl/span

```diff
@@ -521,23 +551,25 @@ public:
         Expects(size() > 0);
         return data()[size() - 1];
     }
-    
-    // at and operator() are deprecated to align to the public member functions of std::span
-    [[deprecated("Use operator[]")]]
-    constexpr reference at(index_type idx) const noexcept{ return this->operator[](idx); }
-    [[deprecated("Use operator[]")]]
-    constexpr reference operator()(index_type idx) const noexcept{ return this->operator[](idx); }
-
 
+    // at and operator() are deprecated to align to the public member functions of std::span
+    [[deprecated("Use operator[]")]] constexpr reference at(index_type idx) const noexcept
+    {
+        return this->operator[](idx);
+    }
+    [[deprecated("Use operator[]")]] constexpr reference operator()(index_type idx) const noexcept
+    {
+        return this->operator[](idx);
+    }
 
     constexpr pointer data() const noexcept { return storage_.data(); }
 
     // [span.iter], span iterator support
-    constexpr iterator begin() const noexcept { return {this, 0}; }
-    constexpr iterator end() const noexcept { return {this, size()}; }
+    constexpr iterator begin() const noexcept { return {data(), data() + size(), data()}; }
+    constexpr iterator end() const noexcept { return {data(), data() + size(), data() + size()}; }
```

Comment:
Debug codegen: Consider factoring out the calls to `data()` and `size()` to avoid calling them repeatedly, as `begin()` and `end()` are popular functions. Ditto for `cbegin()` and `cend()` below.